shell/shell.go

```diff
@@ -44,6 +44,13 @@ func Env(vars map[string]string) RunOption {
 	}
 }
 
+// Dir sets the working directory for the command.
+func Dir(dir string) RunOption {
+	return func(r *interp.Runner) {
+		r.Dir = dir
```

Comment:
should we do this relative to the configured working directory, if it's not an absolute path?

Reply:
This seems like a good thing to do to keep the behaviour predictable. I'll take a look at the code and find the canonical place to retrieve this path from.

Reply:
Seems that if we want to avoid repeatedly re-parsing workpace.json, we have one of two options:
 - create a global `Config` or `Executor` instance that we can arbitrary access at runtime through the program, or:
 - modify `RunOption` so it accepts an `Executor` as well as an `*interp.Runner` (interface-breaking change)

I'm more in favour of the second option, but that would require a major version update because it'd break backward-compatibility.

Do you have any strong opinions or suggestions for an approach?

Reply:
global config doesn't sound too bad if we can get the old and new codepath to use the same logic, even if it's different objects.

i'm somewhat against breaking backwards compat. it probably wouldn't be too hard to move all of internal uses over, but repos outside the monorepo use it as well